event-handling/adding-onclick-handling.md

```diff
@@ -0,0 +1,173 @@
+# Adding onclick Handling
+
+## Goals
+
+Our goals for this lesson is to implement event handling in JavaScript. 
+
+## Introduction
+
+Keira is a web developer creating [a website dedicated to the science and beauty of crabs](https://replit.com/@adacore/Adding-onclick-Handling-Demo#index.html). Keira wants to add crabs to her website every time the user clicks a button named "Add Crab."
+
+Keira will need to do two things every time she wants to handle an event:
+
+1. Build an event handler
+1. Register the event handler to some HTML element(s)
+
+## Vocabulary
+
+| Vocab         | Definition                                                                                                           | How to Use in a Sentence                                                                                                                                                                                                            |
+| ------------- | -------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Event Handler | A function whose responsibility is to perform necessary, required, or desired actions after a certain event is fired | "The function is a click event handler because we expect it to run after the button is clicked," "In order for something to happen after we scroll, we need to make an event handler for what should happen when the user scrolls." |
+
```

Comment:
## Preparing the Page

Before we look at how to set up our event handler, we need to have an element that can raise the events we want to receive. We should aso set up any other elements that we will modify in response to the event.

We know that we need a button named "Add Crab", so we add the following HTML to our page:

```html
<section>
  <button id="addCrabButton">Add Crab</button>
</section>
```

The `<section>` is optional, but it can be useful to wrap controls in another grouping element to help with styling.

Next, we need to make sure there's place for us to add all those crabs! We can add another section to act as a container for them all.

```html
<section id="crabContainer"></section>
```

And that's it! Moving forward, we'll look at how to listen for events coming from the button, and then add lots and lots of crabs to crab container using an event handler!